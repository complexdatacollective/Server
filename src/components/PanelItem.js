import React from 'react';
import PropTypes from 'prop-types';

const PanelItem = ({ label, value }) => (
  <div className="panel-item">
    <div className="panel-item__label">{label}</div>
    <div className="panel-item__value">
      {value}
    </div>
  </div>
);

PanelItem.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};

export default PanelItem;
