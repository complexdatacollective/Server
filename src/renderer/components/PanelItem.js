import React from 'react';
import PropTypes from 'prop-types';

const PanelItem = ({ label, placeholder, value }) => (
  <div className="panel-item">
    <div className="panel-item__label">{label}</div>
    {
      value && <div className="panel-item__value">{value}</div>
    }
    {
      !value && <div className="panel-item__placeholder">{ placeholder }</div>
    }
  </div>
);

PanelItem.defaultProps = {
  placeholder: '',
  value: null,
};

PanelItem.propTypes = {
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};

export default PanelItem;
