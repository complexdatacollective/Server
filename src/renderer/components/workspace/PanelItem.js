import React from 'react';
import PropTypes from 'prop-types';

const renderItem = (item) => <div key={item} className="panel-item__value">{item}</div>;

const renderValue = (value, placeholder) => {
  if (Array.isArray(value)) {
    return value.map((item) => renderItem(item));
  }
  if (value) {
    return renderItem(value);
  }
  return (<div className="panel-item__placeholder">{ placeholder }</div>);
};

const PanelItem = ({ label, placeholder, value }) => (
  <div className="panel-item">
    <div className="panel-item__label">{label}</div>
    {
      renderValue(value, placeholder)
    }
  </div>
);

PanelItem.defaultProps = {
  placeholder: '',
  value: null,
};

PanelItem.propTypes = {
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.array,
  ]),
};

export default PanelItem;
