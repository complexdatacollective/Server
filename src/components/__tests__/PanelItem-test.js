/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import PanelItem from '../PanelItem';

const mockProps = {
  label: 'foo',
  value: 'bar',
};

describe('<PanelItem />', () => {
  it('should render', () => {
    const subject = shallow((
      <PanelItem {...mockProps} />
    ));

    expect(subject).toMatchSnapshot();
  });
});
