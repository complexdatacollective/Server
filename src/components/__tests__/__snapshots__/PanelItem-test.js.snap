// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PanelItem /> should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="panel-item"
>
    <div
        className="panel-item__label"
    >
        foo
    </div>
    <div
        className="panel-item__value"
    >
        bar
    </div>
</div>,
  "nodes": Array [
    <div
      className="panel-item"
>
      <div
            className="panel-item__label"
      >
            foo
      </div>
      <div
            className="panel-item__value"
      >
            bar
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <PanelItem
        label="foo"
        value="bar"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "label": "foo",
          "value": "bar",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="panel-item"
>
          <div
                    className="panel-item__label"
          >
                    foo
          </div>
          <div
                    className="panel-item__value"
          >
                    bar
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="panel-item"
>
          <div
                    className="panel-item__label"
          >
                    foo
          </div>
          <div
                    className="panel-item__value"
          >
                    bar
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PanelItem
    label="foo"
    value="bar"
/>,
}
`;
